pepper-apis/src/main/java/org/broadinstitute/ddp/db/dao/PicklistQuestionCachedDao.java

```diff
@@ -56,33 +57,37 @@ public TemplateDao getTemplateDao() {
 
     @Override
     public Map<Long, GroupAndOptionDtos> findOrderedGroupAndOptionDtos(Iterable<Long> questionIds, long timestamp) {
-        if (isNullCache(questionIdToGroupAndOptionsCache)) {
+        if (isNullCache(questionVersionKeyToGroupAndOptionsCache)) {
             return delegate.findOrderedGroupAndOptionDtos(questionIds, timestamp);
         } else {
             Set<Long> missingQuestionIds = new HashSet<>();
             Map<Long, GroupAndOptionDtos> result = new HashMap<>();
 
             for (var questionId : questionIds) {
+                String key = questionId + ":" + timestamp;
```

Comment:
Key is made by joining with a `:` (colon) separator. Other `*CachedDao` classes uses colon so we're just following convention here.